pkg/sql/colexec/join/join.go

```diff
@@ -105,11 +105,12 @@ func (ctr *container) probe(bat *batch.Batch, ap *Argument, proc *process.Proces
 	anal.Input(bat, isFirst)
 	rbat := batch.NewWithSize(len(ap.Result))
 	rbat.Zs = proc.Mp().GetSels()
+	preAllocLen := bat.Length()
```

Comment:
Would setting it directly to bat.Length() be too much of a memory waste if the number of lines matched is only 1?